File: mobile_ui/quickchat_ui/lib/features/authentication/domain/repositories/auth_repository.dart
import 'package:dartz/dartz.dart';
import 'package:quickchat_ui/core/logic/result.dart';
import 'package:quickchat_ui/features/authentication/data/dto/LoginData.dto.dart';
import 'package:quickchat_ui/features/authentication/data/dto/LoginUser.dto.dart';
import 'package:quickchat_ui/features/authentication/data/dto/RegisterUser.dto.dart';

abstract class AuthRepository {
  Future<Result<Unit>> connectServer();
  Future<Result<LoginUserDto>> getCredentials();
  Future<Result<Unit>> registerUser(
    RegisterUserDto registerUserDto,
  );
  Future<Result<LoginDatasDto>> loginUser(
    LoginUserDto loginUserDto,
  );
  Future<Result<LoginDatasDto>> autoLoginUser();
  Future<Result<Unit>> logoutUser();
}

File: web_ui/app/features/authentication/domain/repositories/auth_repository.ts
import { Result } from "@/app/core/result";
import { LoginUserDto } from "../../data/dto/LoginUser.dto";
import { LoginDatasDto } from "../../data/dto/LoginData.dto";
import { RegisterUserDto } from "../../data/dto/RegisterUser.dto";

export interface AuthRepository {
  loginUser(loginUserDto: LoginUserDto): Promise<Result<LoginDatasDto>>;
  autoLogin(): Promise<Result<LoginDatasDto>>;
  registerUser(registerUserDto: RegisterUserDto): Promise<Result<void>>;
  logoutUser(): Promise<Result<void>>;
}
